### PR TITLE
feat(client): cypress

### PR DESCRIPTION
**Pull request by momentiris**
#23 

Adds cypress testing suite sprinkled with cucumber syntax.

Try it with `npm run cypress`

### CODE REVIEW

File: packages/client/package.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
     "test": "is-ci \"test:ci\" \"test:watch\"",
     "test:ci": "jest",
     "test:watch": "jest --watch",
+    "buildAndServe": "npm run webpack:production && npm run serve",
+    "serve": "serve ./dist -l 3001 -s",
+    "cypress": "start-server-and-test buildAndServe http://localhost:3001 cypress:gui",
```
Comment (lessp) [inline packages/client/package.json]: `start-server-and-test` is not installed!

`npm i -D start-server-and-test`

Comment (momentiris) [reply packages/client/package.json]: What do you mean? It's already a dev dependency.

Comment (lessp) [reply packages/client/package.json]: Ah, sorry! Must have forgot to run `npm i`!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+**Ticket:** [#](TRELLOURL)
+**Intent:**
+**How to test (optional):**
+
+### All of the following commands should pass.
+
+- [ ] `npm test`
```
Comment (lessp) [inline .github/PULL_REQUEST_TEMPLATE.md]: I wonder if it's more convenient to add some pre-commit-hooks and/or set up Github actions/Travis as a separate task for this instead?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+**Ticket:** [#](TRELLOURL)
```
Comment (lessp) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
**Issue:** Closes #
```

File: packages/client/src/components/Login.re [hunk 4]
```diff
@@ -94,6 +94,7 @@ let make = (~onLogin) => {
             "Username"->React.string
           </label>
           <input
+            ref={UseAutoFocus.use()}
```
Comment (lessp) [inline packages/client/src/components/Login.re]: Does this feel a bit off? Perhaps it's clearer what it does if we use it closer to the top?

```ocaml
let usernameInputRef = UseAutoFocus.use();

...

<input ref=usernameInputRef .../>
```

File: packages/client/src/components/Login.re [hunk 5]
```diff
@@ -119,10 +120,11 @@ let make = (~onLogin) => {
         <input
```
Comment (lessp) [inline packages/client/src/components/Login.re]: I know this is not a part of this PR and I can do it after it gets merged, but I think using a `<button>`-element here would be nice so one doesn't have to look for the `type` to see that this is submitting!

File: packages/client/src/hooks/UseAutoFocus.re [hunk 6]
```diff
@@ -0,0 +1,18 @@
+let use = () => {
```
Comment (lessp) [inline packages/client/src/hooks/UseAutoFocus.re]: Nice! 💯

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (lessp) [reply .github/PULL_REQUEST_TEMPLATE.md]: If so, might as well remove this file altogether?

**Pull request merged**